dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -603,6 +605,189 @@ def test_correlation_update(self):
                                              profiler.correlation_matrix)
 
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler',
+                spec=StructuredDataLabeler)
+    def test_chi2(self, *mocks):
+        # Empty
+        data = pd.DataFrame([])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.chi2_matrix)
+
+        # Single column
+        data = pd.DataFrame({'a': ["y", "y", "n", "n", "y"]})
+        profiler = dp.StructuredProfiler(data)
+        expected_mat = np.array([1])
+        self.assertEqual(expected_mat, profiler.chi2_matrix)
+
+        data = pd.DataFrame({'a': ["y", "y", "y", "y", "n", "n", "n"],
+                             'b': ["y", "maybe", "y", "y", "n", "n", "maybe"],
+                             'c': ["n", "maybe", "n", "n", "n", "y", "y"]})
+
+        profiler = dp.StructuredProfiler(data)
+        expected_mat = np.array([
+            [1, 0.309924, 0.404638],
+            [0.309924, 1, 0.548812],
+            [0.404638, 0.548812, 1]
+        ])
+        np.testing.assert_array_almost_equal(expected_mat,
+                                             profiler.chi2_matrix)
+
+        # All different categories
+        data = pd.DataFrame({'a': ["y", "y", "y", "y", "n", "n", "n"],
+                             'b': ["a", "maybe", "a", "a", "b", "b", "maybe"],
+                             'c': ["d", "d", "g", "g", "g", "t", "t"]})
+
+        profiler = dp.StructuredProfiler(data)
+        expected_mat = np.array([
+            [1, 0.007295, 0.007295],
+            [0.007295, 1, 0.015609],
+            [0.007295, 0.015609, 1]
+        ])
+        np.testing.assert_array_almost_equal(expected_mat,
+                                             profiler.chi2_matrix)
+
+        # Identical columns
+        data = pd.DataFrame({'a': ["y", "y", "y", "y", "n", "n", "n"],
+                             'b': ["y", "y", "y", "y", "n", "n", "n"],
+                             'c': ["y", "y", "y", "y", "n", "n", "n"]})
+
+        profiler = dp.StructuredProfiler(data)
+        expected_mat = np.array([
+            [1, 1, 1],
+            [1, 1, 1],
+            [1, 1, 1]
+        ])
+        np.testing.assert_array_almost_equal(expected_mat,
+                                             profiler.chi2_matrix)
+
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler',
+                spec=StructuredDataLabeler)
+    def test_merge_chi2(self, *mocks):
+        data = pd.DataFrame({'a': ["y", "y", "y", "y", "n", "n", "n"],
+                             'b': ["y", "maybe", "y", "y", "n", "n", "maybe"],
+                             'c': ["n", "maybe", "n", "n", "n", "y", "y"]})
+
+        data1 = data[:4]
+        data2 = data[4:]
```

Comment:
can we add merge from 
- 2 empty data
- 1 empty and 1 non-empty

Reply:
our correlation tests do this too, right?

Reply:
Merging doesn't allow for empty data:
```python
# In the case of __add__ with one of the schemas not initialized
if strict and (len_schema1 == 0 or len_schema2 == 0):
    raise ValueError("Cannot merge empty profiles.")
```

Reply:
I think we can use the same logic as with correlation
```
corr_mat1 = self.correlation_matrix
        corr_mat2 = other.correlation_matrix
        n1 = self.total_samples - self.row_is_null_count
        n2 = other.total_samples - other.row_is_null_count
        if n1 == 0:
            return corr_mat2
        if n2 == 0:
            return corr_mat1

        if corr_mat1 is None or corr_mat2 is None:
            return None
```